### PR TITLE
Or filtering UI

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe. If this affects the frontend, include screenshots.*

I couldn't rebase off of master properly on https://github.com/PostHog/posthog/pull/8617 sorry 😬 but here is that PR with most of the comments addressed. This UI addition is missing a couple of things from the mocks by design because there's still a lot of backend stuff going on for it right now, so just going to hold off from polishing this in favor of getting a smaller PR merged.

https://user-images.githubusercontent.com/25164963/156004498-8a31eec3-8c84-43fa-b6a4-eb5f7ba1b0a8.mov

*Stay up-to-date with our [coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Add feature flag `and-or-filtering` and use the trends tab to test

*Briefly describe the steps you took.*

### CODE REVIEW

File: frontend/src/types.ts [hunk 1]
```diff
@@ -1432,6 +1432,20 @@ export interface SelectOption {
     label?: string
 }
 
+export enum FilterLogicalOperator {
+    And = 'AND',
+    Or = 'OR',
+}
+export interface AndOrPropertyFilter {
```
Comment (neilkakkar) [inline frontend/src/types.ts:1439]: The name here doesn't make sense to me. (and seems similar to the one below? Can't tell the difference using the names)

Perhaps call this `PropertyGroupFilter`?

and for the one below, can you explain where / why its used? 🤔

**Review by EDsCODE — COMMENTED** [review]
looking good. Will do a manual QA this afternoon and should be good to merge soon

File: frontend/src/types.ts (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/types.ts:1439]: yep!

Comment (liyiy) [reply frontend/src/types.ts]: ✅

File: frontend/src/lib/components/PropertyFilters/components/taxonomicPropertyFilterLogic.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ export const taxonomicPropertyFilterLogic = kea<taxonomicPropertyFilterLogicType
     selectors: {
         filter: [
             (s) => [s.filters, (_, props) => props.filterIndex],
-            (filters, filterIndex): AnyPropertyFilter | null => filters[filterIndex] || null,
+            (filters, filterIndex) => filters[filterIndex] || null,
```
[the remark this answers is not in the document]
Comment (liyiy) [reply frontend/src/lib/components/PropertyFilters/components/taxonomicPropertyFilterLogic.ts:54]: fixed ✅

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**